libraries/chain/controller.cpp

```diff
@@ -2501,6 +2501,28 @@ struct controller_impl {
             }
          }
 
+         // special case for in-place upgrade of code_index
+         if (std::is_same_v<value_t, code_object>) {
+            using v1 = snapshot_code_object_v1;
+            if (std::clamp(header.version, v1::minimum_version, v1::maximum_version) == header.version ) {
+               snapshot->read_section<code_object>([&db=this->db](auto &section) {
+                  v1 obj_v1;
+                  section.read_row(obj_v1, db);
+                  bool sync_call_supported = wasm_interface::is_sync_call_supported(obj_v1.code.data(), obj_v1.code.size());
```

Comment:
Should we just set `sync_call_supported` to `false` for old snapshots? This way, we will not "accidentally" execute an exported `sync_call` function in old code if it has valid signature, because we don't know if the code owner wants the function to be called in the new sync call sense. If the owner wants it to be used by a sync call after the sync call protocol feature is activated, he should redeploy the contract.

But on the other hand, if a function is exported, it is expected to be used in any cases.

Reply:
Turns out we don't need to put `sync_call_supported` into snapshots. OK to re-calculate it when a snapshot is read.